From 476575499d92457a92f1cd7c573d538e45166c5b Mon Sep 17 00:00:00 2001
From: "Xu,Randy" <randy.xu@intel.com>
Date: Fri, 9 Dec 2016 10:14:59 +0800
Subject: [PATCH] Solve the EGL configure issue in RenderEngine

After upgrade to HWC2.0, the window surface format is assigned as
PIXEL_FORMAT_RGBA_8888 in DisplayDevice::DisplayDevice, while
RenderEngine::selectEGLConfig doesnt provide Alpha info, the driver
will provide RGBX instead, which will cause CTS failure in
android.display.cts.VirtualDisplayXXX

JIRA: https://01.org/jira/browse/AIA-70
Test: Pass CtsDisplayTestCases and have no regression in other CTS test.

Signed-off-by: Xu,Randy <randy.xu@intel.com>
---
 services/surfaceflinger/RenderEngine/RenderEngine.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/services/surfaceflinger/RenderEngine/RenderEngine.cpp b/services/surfaceflinger/RenderEngine/RenderEngine.cpp
index d6a032f..ce1e63b 100644
--- a/services/surfaceflinger/RenderEngine/RenderEngine.cpp
+++ b/services/surfaceflinger/RenderEngine/RenderEngine.cpp
@@ -380,6 +380,8 @@ static status_t selectEGLConfig(EGLDisplay display, EGLint format,
         attribs[EGL_RED_SIZE]                   = 8;
         attribs[EGL_GREEN_SIZE]                 = 8;
         attribs[EGL_BLUE_SIZE]                  = 8;
+        if(format == HAL_PIXEL_FORMAT_RGBA_8888)
+                attribs[EGL_ALPHA_SIZE]         = 8;
         wantedAttribute                         = EGL_NONE;
         wantedAttributeValue                    = EGL_NONE;
     } else {
--
2.7.4
